pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstance.java

```diff
@@ -49,14 +54,11 @@ public JavaInstance(InstanceConfig config, Object userClassObject,
                  PulsarClient pulsarClient,
                  Source source) {
         // TODO: cache logger instances by functions?
-        Logger instanceLog = LoggerFactory.getLogger("function-" + config.getFunctionDetails().getName());
-
-        if (source instanceof PulsarSource) {
```

Comment:
Why did we get rid of this logic? We need to set ContextImpl accordingly

Reply:
Look a little closer. it's now under handleMessage. We're building N consumers now and we need to set the appropriate Consumer in the context.

Reply:
We should try to create  a new ContextImpl object in handle message since that is called on every message.  This may incur high GC costs